dev/WebView2/WebView2.cpp

```diff
@@ -0,0 +1,1932 @@
+
+            POINT coords_win32;
+            POINTSTOPOINT(coords_win32, l_param);
+            winrt::Point coords{ static_cast<float>(coords_win32.x), static_cast<float>(coords_win32.y) };
+
+            // mouse data is nonzero for mouse wheel scrolling and XBUTTON events
+            DWORD mouse_data = 0;
+            if (message == WM_MOUSEWHEEL || message == WM_MOUSEHWHEEL)
+            {
+                mouse_data = GET_WHEEL_DELTA_WPARAM(w_param);
+            }
+            else if (message == WM_XBUTTONDOWN || message == WM_XBUTTONUP || message == WM_XBUTTONDBLCLK)
+            {
+                mouse_data = GET_XBUTTON_WPARAM(w_param);
+            }
+
+            m_coreWebViewCompositionController.SendMouseInput(
+                winrt::CoreWebView2MouseEventKind{ static_cast<winrt::CoreWebView2MouseEventKind>(message) },
+                winrt::CoreWebView2MouseEventVirtualKeys{ static_cast<winrt::CoreWebView2MouseEventVirtualKeys>(GET_KEYSTATE_WPARAM(w_param)) },
+                mouse_data,
+                coords);
+        }
+    }
+    else if ((deviceType == winrt::Windows::Devices::Input::PointerDeviceType::Touch) ||
+             (deviceType == winrt::Windows::Devices::Input::PointerDeviceType::Pen))
+    {
+        const winrt::PointerPoint inputPt{ args.GetCurrentPoint(*this) };
+        winrt::CoreWebView2PointerInfo outputPt = m_coreWebViewEnvironment.CreateCoreWebView2PointerInfo();
+
+        //PEN INPUT
+        if (deviceType == winrt::PointerDeviceType::Pen)
+        {
+            FillPointerPenInfo(inputPt, outputPt);
+        }
+
+        //TOUCH INPUT
+        if (deviceType == winrt::PointerDeviceType::Touch)
+        {
+            FillPointerTouchInfo(inputPt, outputPt);
+        }
+
+        //GENERAL POINTER INPUT
+        FillPointerInfo(inputPt, outputPt, args);
+
+        m_coreWebViewCompositionController.SendPointerInput(winrt::CoreWebView2PointerEventKind{ static_cast<winrt::CoreWebView2PointerEventKind>(message) }, outputPt);
+    }
+}
+
+// The transform is not available in matrix form outside core windows so needed
+// information about the transformation needs to be reconstructed by applying
+// the transform directly to a known set of points.
+// It is assumed that no shear transform is applied and currently rotation is not supported.
+winrt::float4x4 WebView2::GetMatrixFromTransform() {
+    // Calculate transformation assuming 2D only.
+    // Calculate transformed values
+    auto generalTransform = TransformToVisual(nullptr);
+    winrt::Point initialOrigin = winrt::Point(0, 0);
+    winrt::Point translatedOrigin = generalTransform.TransformPoint(initialOrigin);
+
+    winrt::float4x4 outputMatrix{};
+
+    // Assign rotation
+    outputMatrix.m12 = 0.0f;
+    outputMatrix.m13 = 0.0f;
+    outputMatrix.m21 = 0.0f;
+    outputMatrix.m23 = 0.0f;
+    outputMatrix.m31 = 0.0f;
+    outputMatrix.m32 = 0.0f;
```

Comment:
these should be 0 inited already anyway

Reply:
This function will be removed (see my comment below).